nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/encrypt/StringEncryptor.java

```diff
@@ -408,27 +468,34 @@ public String decrypt(String cipherText) throws EncryptionException {
 
     private byte[] decryptPBE(byte[] cipherBytes) {
         PBECipherProvider pbecp = (PBECipherProvider) cipherProvider;
-        final EncryptionMethod encryptionMethod = EncryptionMethod.forAlgorithm(algorithm);
+        final EncryptionMethod encryptionMethod = getEncryptionMethodForAlgorithm(algorithm);
 
         // Extract salt
-        int saltLength = CipherUtility.getSaltLengthForAlgorithm(algorithm);
+        int saltLength = determineSaltLength(algorithm);
         byte[] salt = new byte[saltLength];
         System.arraycopy(cipherBytes, 0, salt, 0, saltLength);
 
-        byte[] actualCipherBytes = Arrays.copyOfRange(cipherBytes, saltLength, cipherBytes.length);
+        // Read IV if necessary (allows for future use of Argon2, PBKDF2, Bcrypt, or Scrypt)
+        byte[] ivBytes = new byte[0];
```

Comment:
maybe it isn't worth the effort.  by preallocating a 16 byte array to reuse over and over seems like it would perform quite well and relax GC pressure particularly in the event of larger objects.  am I overthinking that one?

Reply:
The actual `ivBytes` array is only instantiated when the algorithm dictates a random IV. For 99.9% of users, this won't be the case and so it makes sense not to instantiate it at all. 

For users who select one of the custom algorithms, the `byte[16]` is instantiated and filled with a random value, but this happens every time and is intentionally unique on each invocation, so pre-allocating wouldn't gain anything unless I am missing your concern. 

The size of the object to be encrypted is independent of the IV size -- the IV is always 0 or 16 bytes depending solely on the algorithm.

Reply:
The instantiation of the `byte[0]` is just to simplify the code which concatenates all the byte arrays below -- rather than having duplicate logic for IV present/not, it just appends the array whether it is populated or empty.